include/seqan3/search/kmer_index/shape.hpp

```diff
@@ -0,0 +1,106 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Enrico Seiler <enrico.seiler AT fu-berlin.de>
+ * \author Vinzenz May <vinzenz.may AT fu-berlin.de>
+ * \brief Provides seqan3::shape.
+ */
+
+#pragma once
+
+#include <array>
+
+#include <seqan3/std/concepts>
+
+namespace seqan3
+{
+
+/*!\brief A class that defines what positions of a pattern to hash.
+*
+* When hashing a sequence, there may be positions that do not count towards the final hash value.
+* A shape offers an easy way to define such patterns. Given a k-mer length `k` (0 < `k` <= 32), a shape
+* represents a binary sequence where a `0` encodes a "don't care position", i.e. a position that is not taken into
+* account when computing the hash value. A `1` therefore translates to a position that is used to compute the hash
+* value.
+*
+* ### Example
+*
+* \include test/snippet/search/kmer_index/shape.cpp
+*
+* \attention 0 < size <= 32
+*/
+
+class shape : public std::array<bool, 32>
+{
+private:
+    //!\brief The size of the shape.
+    size_t k;
+
+public:
+
+    using std::array<bool, 32>::array;
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    shape()                                    noexcept = default; //!< Defaulted.
+    constexpr shape(shape const &)             noexcept = default; //!< Defaulted.
+    constexpr shape(shape &&)                  noexcept = default; //!< Defaulted.
+    constexpr shape & operator=(shape const &) noexcept = default; //!< Defaulted.
+    constexpr shape & operator=(shape &&)      noexcept = default; //!< Defaulted.
+    ~shape()                                   noexcept = default; //!< Defaulted.
+
+    /*!\brief Construct from a given size.
```

Comment:
```suggestion
    /*!\brief Construct an ungapped shape from a given size.
```
Or add in the brief section that this will fill the shape with "every position is important" or somehting like that.